kernel/safe_typing.ml

```diff
@@ -328,16 +328,20 @@ let env_of_senv = env_of_safe_env
 
 let sections_of_safe_env senv = senv.sections
 
+let get_section = function
+  | None -> CErrors.user_err Pp.(str "No open section.")
+  | Some s -> s
```

Comment:
[nit] It seems that the users below follow the same pattern to an extent that it could be worth to make this into a combinator `with_open_section` and refactor a bit more.